aiidalab/app.py

```diff
@@ -172,11 +173,19 @@ def _install_dependencies(self, python_bin):
         """Try to install the app dependencies with pip (if specified)."""
 
         def _pip_install(*args):
-            run(
+            process = subprocess.Popen(
                 [python_bin, "-m", "pip", "install", *args],
-                check=True,
                 encoding="utf-8",
+                bufsize=1,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
             )
+            while True:
+                output = process.stdout.readline()
+                if output == "" and process.poll() is not None:
+                    break
+                if output:
+                    self.stdout.write(output)
```

Comment:
Is this a recipe from SO or so? If yes, could you please add a comment with the link?

Reply:
added comment.